eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/Bag.java

```diff
@@ -214,6 +214,14 @@
 
     ImmutableBagIterable<T> toImmutable();
 
+    /**
+     * Returns an unmodifiable view on the distinct elements with the same complexity as the Bag implementation.
+     *
+     * @return an unmodifiable view on the distinct elements of the Bag.
+     * @since 11.1
+     */
+    RichIterable<T> distinctView();
```

Comment:
Hi @victornoel, this method will need a default implementation in order to be included in the 11.1 release. The simplest option would be to throw an `UnsupportedOperationException` with a message that this must be implemented in subclasses.